src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,90 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] map;
+
+    public MyHashMap() {
+        this.map = new Node[DEFAULT_CAPACITY];
+        this.threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold) {
+            resize();
+        }
+        int nodeIndex = getNodeIndex(key);
+        Node<K, V> node = searchNode(key, nodeIndex);
+        if (node != null) {
+            node.value = value;
+        } else {
+            Node<K, V> newNode = new Node<>(key, value, map[nodeIndex]);
+            map[nodeIndex] = newNode;
+            size++;
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> node = searchNode(key, getNodeIndex(key));
+        return node != null ? node.value : null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private static class Node<K, V> {
+        private final K key;
+        private V value;
+        private Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    private void resize() {
+        int stepIncreaseCapacity = 2;
+        Node<K, V>[] newMap = new Node[DEFAULT_CAPACITY * stepIncreaseCapacity];
```

Comment:
I think you could do this in one line, without creating `int stepIncreaseCapacity`